src/query/streams.cpp

```diff
@@ -185,7 +186,27 @@ Streams::Streams(InterpreterContext *interpreter_context, std::string bootstrap_
                  std::filesystem::path directory)
     : interpreter_context_(interpreter_context),
       bootstrap_servers_(std::move(bootstrap_servers)),
-      storage_(std::move(directory)) {}
+      storage_(std::move(directory)) {
+  constexpr std::string_view proc_name = "kafka_set_stream_offset";
+  auto set_stream_offset = [ictx = interpreter_context, proc_name](mgp_list *args, mgp_graph * /*graph*/,
+                                                                   mgp_result *result, mgp_memory * /*memory*/) {
+    auto *arg_stream_name = procedure::Call<mgp_value *>(mgp_list_at, args, 0);
+    const auto *stream_name = procedure::Call<const char *>(mgp_value_get_string, arg_stream_name);
+    auto *arg_offset = procedure::Call<mgp_value *>(mgp_list_at, args, 1);
+    const auto offset = procedure::Call<int64_t>(mgp_value_get_int, arg_offset);
+    const auto error = ictx->streams.SetStreamOffset(stream_name, offset);
+    if (error.HasError()) {
+      MG_ASSERT(mgp_result_set_error_msg(result, error.GetError().c_str()) == MGP_ERROR_NO_ERROR,
+                "{} unable to set stream offset", proc_name);
```

Comment:
```suggestion
                "Unable to set procedure error message");
```
In my opinion this assert fails when we cannot set the error message, so the current message might be misleading for the users: it suggest that memgraph crashed because we were unable to set stream offset.